### ISSUE TITLE
When failed to create a job, engine's status is always "Created'

### ISSUE THREAD

**Issue by lance6716** [pingcap/tiflow#6915]
### What did you do?

```
# lance @ lance6716-nuc10i7fnh in /tmp/tiflow_engine_test [16:38:59] 
$ curl http://127.0.0.1:10245/api/v1/jobs/d2fb3ff4-a357-411d-ad45-d1063bcc13d3 

{"id":"d2fb3ff4-a357-411d-ad45-d1063bcc13d3", "type":"DM", "status":"Created", "config":"", "error":null}

                                                     
# lance @ lance6716-nuc10i7fnh in /tmp/tiflow_engine_test [16:49:28] 
$ curl http://127.0.0.1:10245/api/v1/jobs/d2fb3ff4-a357-411d-ad45-d1063bcc13d3/status

{"code":9, "details":[{"@type":"type.googleapis.com/enginepb.ErrorV2", "name":"JobNotRunningError", "details":"eyJKb2JJRCI6ImQyZmIzZmY0LWEzNTctNDExZC1hZDQ1LWQxMDYzYmNjMTNkMyJ9", "stack_trace":"\ngithub.com/pingcap/tiflow/engine/pkg/rpcerror.(*Prototype[...]).GenWithStack\n\tgithub.com/pingcap/tiflow/engine/pkg/rpcerror/normalize.go:101\ngithub.com/pingcap/tiflow/engine/servermaster.(*JobManagerImpl).GetJobMasterForwardAddress\n\tgithub.com/pingcap/tiflow/engine/servermaster/jobmanager.go:365\ngithub.com/pingcap/tiflow/engine/servermaster.(*Server).handleForwardJobAPI\n\tgithub.com/pingcap/tiflow/engine/servermaster/server.go:735\ngithub.com/pingcap/tiflow/engine/servermaster.(*Server).forwardJobAPI\n\tgithub.com/pingcap/tiflow/engine/servermaster/server.go:708\ngithub.com/pingcap/tiflow/engine/servermaster.registerRoutes.func1\n\tgithub.com/pingcap/tiflow/engine/servermaster/http.go:48\nnet/http.HandlerFunc.ServeHTTP\n\tnet/http/server.go:2109\nnet/http.(*ServeMux).ServeHTTP\n\tnet/http/server.go:2487\nnet/http.serverHandler.ServeHTTP\n\tnet/http/server.go:2947\nnet/http.(*conn).serve\n\tnet/http/server.go:1991\nruntime.goexit\n\truntime/asm_amd64.s:1594"}]}%     
```

server-master.log
```
[2022/08/29 08:49:08.280 +00:00] [INFO] [master.go:669] ["DispatchTask failed"] [job_id=dataflow-engine-job-manager] [error="rpc error: code = Aborted desc = [code=20019:class=config:scope=internal:level=medium], Message: must specify a unique task name, Workaround: Please check the `name` config in task configuration file."] [errorVerbose="rpc error: code = Aborted desc = [code=20019:class=config:scope=internal:level=medium], Message: must specify a unique task name, Workaround: Please check the `name` config in task configuration file.\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/engine/pkg/rpcerror.FromGRPCError\n\tgithub.com/pingcap/tiflow/engine/pkg/rpcerror/grpc.go:71\ngithub.com/pingcap/tiflow/engine/pkg/client/internal.(*Call[...]).callOnce\n\tgithub.com/pingcap/tiflow/engine/pkg/client/internal/call.go:88\ngithub.com/pingcap/tiflow/engine/pkg/client/internal.(*Call[...]).Do.func1\n\tgithub.com/pingcap/tiflow/engine/pkg/client/internal/call.go:74\ngithub.com/pingcap/tiflow/pkg/retry.run\n\tgithub.com/pingcap/tiflow/pkg/retry/retry_with_opt.go:57\ngithub.com/pingcap/tiflow/pkg/retry.Do\n\tgithub.com/pingcap/tiflow/pkg/retry/retry_with_opt.go:34\ngithub.com/pingcap/tiflow/engine/pkg/client/internal.(*Call[...]).Do\n\tgithub.com/pingcap/tiflow/engine/pkg/client/internal/call.go:72\ngithub.com/pingcap/tiflow/engine/pkg/client.(*executorServiceClient).DispatchTask\n\tgithub.com/pingcap/tiflow/engine/pkg/client/executor_service_client.go:96\ngithub.com/pingcap/tiflow/engine/framework.(*DefaultBaseMaster).CreateWorker.func1\n\tgithub.com/pingcap/tiflow/engine/framework/master.go:661\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
```

executor.log
```
[2022/08/29 08:49:08.769 +00:00] [ERROR] [server.go:212] ["Failed to create worker"] [error="[code=20019:class=config:scope=internal:level=medium], Message: must specify a unique task name, Workaround: Please check the `name` config in task configuration file."] [errorVerbose="[code=20019:class=config:scope=internal:level=medium], Message: must specify a unique task name, Workaround: Please check the `name` config in task configuration file.\ngithub.com/pingcap/tiflow/dm/pkg/terror.(*Error).Generate\n\tgithub.com/pingcap/tiflow/dm/pkg/terror/terror.go:271\ngithub.com/pingcap/tiflow/dm/config.(*TaskConfig).adjust\n\tgithub.com/pingcap/tiflow/dm/config/task.go:584\ngithub.com/pingcap/tiflow/dm/config.(*TaskConfig).Adjust\n\tgithub.com/pingcap/tiflow/dm/config/task.go:579\ngithub.com/pingcap/tiflow/engine/jobmaster/dm/config.(*JobCfg).adjust\n\tgithub.com/pingcap/tiflow/engine/jobmaster/dm/config/config.go:229\ngithub.com/pingcap/tiflow/engine/jobmaster/dm/config.(*JobCfg).Decode\n\tgithub.com/pingcap/tiflow/engine/jobmaster/dm/config/config.go:141\ngithub.com/pingcap/tiflow/engine/jobmaster/dm.dmJobMasterFactory.DeserializeConfig\n\tgithub.com/pingcap/tiflow/engine/jobmaster/dm/dm_jobmaster.go:81\ngithub.com/pingcap/tiflow/engine/framework/registry.(*registryImpl).CreateWorker\n\tgithub.com/pingcap/tiflow/engine/framework/registry/registry.go:94\ngithub.com/pingcap/tiflow/engine/executor.(*Server).makeTask\n\tgithub.com/pingcap/tiflow/engine/executor/server.go:203\ngithub.com/pingcap/tiflow/engine/executor.(*Server).PreDispatchTask\n\tgithub.com/pingcap/tiflow/engine/executor/server.go:229\ngithub.com/pingcap/tiflow/engine/enginepb._ExecutorService_PreDispatchTask_Handler\n\tgithub.com/pingcap/tiflow/engine/enginepb/executor_grpc.pb.go:87\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\tgoogle.golang.org/grpc@v1.48.0/server.go:1295\ngoogle.golang.org/grpc.(*Server).handleStream\n\tgoogle.golang.org/grpc@v1.48.0/server.go:1636\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\tgoogle.golang.org/grpc@v1.48.0/server.go:932\nruntime.goexit\n\truntime/asm_amd64.s:1594\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/engine/framework/registry.(*registryImpl).CreateWorker\n\tgithub.com/pingcap/tiflow/engine/framework/registry/registry.go:96\ngithub.com/pingcap/tiflow/engine/executor.(*Server).makeTask\n\tgithub.com/pingcap/tiflow/engine/executor/server.go:203\ngithub.com/pingcap/tiflow/engine/executor.(*Server).PreDispatchTask\n\tgithub.com/pingcap/tiflow/engine/executor/server.go:229\ngithub.com/pingcap/tiflow/engine/enginepb._ExecutorService_PreDispatchTask_Handler\n\tgithub.com/pingcap/tiflow/engine/enginepb/executor_grpc.pb.go:87\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\tgoogle.golang.org/grpc@v1.48.0/server.go:1295\ngoogle.golang.org/grpc.(*Server).handleStream\n\tgoogle.golang.org/grpc@v1.48.0/server.go:1636\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\tgoogle.golang.org/grpc@v1.48.0/server.go:932\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
```

### What did you expect to see?

should be Error

### What did you see instead?

see above

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

```console
(paste Dataflow Engine version here)
```

**Comment by maxshuang** [comment]
We will add the `max-error-retry-times` to jobmanager later to change the job state to `Failed`.

**Comment by amyangfei** [comment]
Will be solved in https://github.com/pingcap/tiflow/issues/6749 , I’m working on it.

**Comment by maxshuang** [comment]
@amyangfei  Can we close this issue now?

**Comment by amyangfei** [comment]
Yes, this scenario is fixed, there still exists several corner cases and will be tracked in #6749

**Comment by lance6716** [comment]
still not fixed at `a8f91d0bb`

```
$ curl http://127.0.0.1:10245/api/v1/jobs/a8feada0-eba1-4e56-80c2-25934fd15bb7 
{"id":"a8feada0-eba1-4e56-80c2-25934fd15bb7", "type":"DM", "state":"Created", "config":"dGFzay1tb2RlOiBmdWxsCnRhcmdldC1kYXRhYmFzZToKICBob3N0OiBob3N0LmRvY2tlci5pbnRlcm5hbAogIHBvcnQ6IDQwMDAKICB1c2VyOiByb290CiAgcGFzc3dvcmQ6ICcnCnVwc3RyZWFtczoKICAtIGRiLWNvbmZpZzoKICAgICAgaG9zdDogaG9zdC5kb2NrZXIuaW50ZXJuYWwKICAgICAgcG9ydDogMzMwNgogICAgICB1c2VyOiBkbV9mdWxsCiAgICAgIHBhc3N3b3JkOiAnMTIzNDU2JwogICAgc291cmNlLWlkOiBteXNxbC0wMQogICAgYmxvY2stYWxsb3ctbGlzdDogYmFsaXN0LTAxCiAgICByb3V0ZS1ydWxlczogWyAiZ2xvYmFsIiBdCmJsb2NrLWFsbG93LWxpc3Q6CiAgYmFsaXN0LTAxOgogICAgZG8tZGJzOgogICAgICAtIGRtX2Z1bGwqCiNyb3V0ZXM6CiMgIGdsb2JhbDoKIyAgICBzY2hlbWEtcGF0dGVybjogImRtX2Z1bGxfcm91dGVfc2NoZW1hIgojICAgIHRhcmdldC1zY2hlbWE6ICJkbV9mdWxsX3JvdXRlX3NjaGVtYV9zdWNjIgo=", "detail":"", "error":{"code":"None", "message":"", "not_leader":null}, "selectors":[]}%  
```

```
$ curl http://127.0.0.1:10245/api/v1/jobs/a8feada0-eba1-4e56-80c2-25934fd15bb7/status
{"code":9, "details":[{"@type":"type.googleapis.com/enginepb.ErrorV2", "name":"JobNotRunningError", "details":"eyJKb2JJRCI6ImE4ZmVhZGEwLWViYTEtNGU1Ni04MGMyLTI1OTM0ZmQxNWJiNyJ9", "stack_trace":"\ngithub.com/pingcap/tiflow/engine/pkg/rpcerror.(*Prototype[...]).GenWithStack\n\tgithub.com/pingcap/tiflow/engine/pkg/rpcerror/normalize.go:101\ngithub.com/pingcap/tiflow/engine/servermaster.(*JobManagerImpl).GetJobMasterForwardAddress\n\tgithub.com/pingcap/tiflow/engine/servermaster/jobmanager.go:480\ngithub.com/pingcap/tiflow/engine/servermaster.(*Server).handleForwardJobAPI\n\tgithub.com/pingcap/tiflow/engine/servermaster/server.go:754\ngithub.com/pingcap/tiflow/engine/servermaster.(*Server).forwardJobAPI\n\tgithub.com/pingcap/tiflow/engine/servermaster/server.go:725\ngithub.com/pingcap/tiflow/engine/servermaster.registerRoutes.func1\n\tgithub.com/pingcap/tiflow/engine/servermaster/http.go:46\nnet/http.HandlerFunc.ServeHTTP\n\tnet/http/server.go:2109\nnet/http.(*ServeMux).ServeHTTP\n\tnet/http/server.go:2487\nnet/http.serverHandler.ServeHTTP\n\tnet/http/server.go:2947\nnet/http.(*conn).serve\n\tnet/http/server.go:1991\nruntime.goexit\n\truntime/asm_amd64.s:1594"}]}%   
```

**Comment by lance6716** [comment]
[diff.txt](https://github.com/pingcap/tiflow/files/9553186/diff.txt)

**Comment by amyangfei** [comment]
Error from `InitImpl` or `Tick` is the second scenario of https://github.com/pingcap/tiflow/pull/6925, it is not fixed yet of coruce. More PRs will be added this week.

**Comment by amyangfei** [comment]
- [ ] Record error correctly https://github.com/pingcap/tiflow/pull/7035
- [ ] Support fast fail on errors returned from Init or Poll.

**Comment by amyangfei** [comment]
Closed by #7061
